Data/task.py

```diff
@@ -1,9 +1,30 @@
+from Data.future import Future
+import api_pb2
+
 class Task:
-    
+    # self.taskData is a list of future or/and bytes
     def __init__(self, taskId, taskDefintion, taskData):
         self.taskId = taskId
         self.taskDefintion = taskDefintion
-        self.taskData = taskData
+        if len(taskData)!=0 and not isinstance(taskData[0], api_pb2.TaskDataParam):
+            self.taskData = taskData
+        # Converts list of api_pb2.TaskDataParam to list of future and/or bytes
+        else:
+            self.taskData = []
+            for i in taskData:
+                if i.WhichOneof("dataParam") == "future":
+                    self.taskData.append(Future(i.future.resultLocation, i.future.hostName, i.future.port))
+                else:
+                    self.taskData.append(i.data)
 
+    def to_proto(self) -> api_pb2.Task:
```

Comment:
could you add a comment about what this function is doing. It is not very easy to understand from code

Reply:
Added